### PR TITLE
Ari parameter binding

### PR DESCRIPTION
**Pull request by AriGlockner**
I finished passing parameters as the environments to functions

### CODE REVIEW

File: interpreter2.rkt [hunk 1]
```diff
@@ -240,6 +234,54 @@
...
+(define (bind-parameters env actual formal return)
+  (cond
+    ; Either the actual or formal parameters is empty
+    ((null? actual)
+     (if (null? formal)
+         (return env)
+         (error "The formal and actual parameters must match")))
+    ((null? formal) (error "The formal and actual parameters must match"))
+    ; Otherwise
+    (else env)))
```
Comment (1ethanhansen) [inline interpreter2.rkt:283]: what is this used for?

File: interpreter2.rkt [hunk 2]
```diff
@@ -240,6 +234,54 @@
...
+          (return binding)
+          (error "The formal and actual parameters must match"))
+      (if (null? actual-param-list)
+          (error "The formal and actual parameters must match")
+          (return (bind-actual-formal-helper env (cdr actual-param-list) (cdr formal-param-list)
+                                             (insert (car formal-param-list) (eval-expression (car actual-param-list) env) binding) return)))))
+
+; Create the environment for the function
+(define (function-environment current-env actual-param-list formal-param-list)
+  (cons (bind-actual-formal current-env actual-param-list formal-param-list)
```
Comment (1ethanhansen) [inline interpreter2.rkt:270]: why does this use cons instead of your newenvironment function?

Comment (AriGlockner) [reply interpreter2.rkt:270]: U right

File: interpreter2.rkt (hunk 1 above)
Comment (AriGlockner) [reply interpreter2.rkt:283]: I'm working on something else to phase this out. This is primarily used for testing purposes

**Comment by 1ethanhansen** [comment]
superseded by #81

**Pull request closed without merging**